pandas/core/tools/datetimes.py

```diff
@@ -343,8 +375,20 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                 # fallback
                 if result is None:
                     try:
-                        result = array_strptime(arg, format, exact=exact,
-                                                errors=errors)
+                        parsing_tzname = '%Z' in format
+                        parsing_tzoffset = '%z' in format
+                        if parsing_tzoffset and parsing_tzname:
+                            raise ValueError("Cannot parse both %Z and %z")
+                        elif tz is not None and (parsing_tzname or
+                                                 parsing_tzoffset):
+                            raise ValueError("Cannot pass a tz argument when "
+                                             "parsing strings with timezone "
+                                             "information.")
+                        result, timezones = array_strptime(
```

Comment:
I would much rather do the error handling in the _return_parsed_timezone_results. This block is just very complicated and hard to grok